libs/remote_storage/src/lib.rs

```diff
@@ -135,35 +171,23 @@ impl std::error::Error for DownloadError {}
 
 /// Every storage, currently supported.
 /// Serves as a simple way to pass around the [`RemoteStorage`] without dealing with generics.
-pub enum GenericRemoteStorage {
-    Local(LocalFs),
-    S3(S3Bucket),
-}
-
-impl GenericRemoteStorage {
-    pub fn new(
-        working_directory: PathBuf,
-        storage_config: &RemoteStorageConfig,
-    ) -> anyhow::Result<Self> {
-        match &storage_config.storage {
-            RemoteStorageKind::LocalFs(root) => {
-                info!("Using fs root '{}' as a remote storage", root.display());
-                LocalFs::new(root.clone(), working_directory).map(GenericRemoteStorage::Local)
-            }
-            RemoteStorageKind::AwsS3(s3_config) => {
-                info!("Using s3 bucket '{}' in region '{}' as a remote storage, prefix in bucket: '{:?}', bucket endpoint: '{:?}'",
-                    s3_config.bucket_name, s3_config.bucket_region, s3_config.prefix_in_bucket, s3_config.endpoint);
-                S3Bucket::new(s3_config, working_directory).map(GenericRemoteStorage::S3)
-            }
+pub type GenericRemoteStorage = Box<dyn RemoteStorage>;
```

Comment:
Frequently we use `Arc<GenericRemoteStorage>` so this ends up being `Arc<Box<dyn RemoteStorage>>`. Arc -> Box becomes a redundant indirection, not that it will hurt, but just to keep things clean

I think this way it worth making `GenericRemoteStorage` actual newtype around `Arc<dyn RemoteStorage>` and implement `Clone` so it calls to `Arc::clone`. Wdyt?

Reply:
Works for me.

You want to take over this PR and rebase / clean it up to make it committable?

Reply:
Sure